theano/__init__.py

```diff
@@ -195,11 +191,11 @@ def sparse_grad(var):
     ret = var.owner.op.__class__(sparse_grad=True)(*var.owner.inputs)
     return ret
 
-# This cannot be done in tensor/__init__.py due to a circular dependency -- randomstreams
-# depends on raw_random which depends on tensor.  As a work-around, we import RandomStreams
-# here and inject an instance in tensor.
-from theano.tensor.randomstreams import RandomStreams
+# This cannot be done in tensor/__init__.py due to a circular
+# dependency -- shared_randomstreams depends on raw_random which
+# depends on tensor.  As a work-around, we import RandomStreams here
+# and inject an instance in tensor.
+from theano.tensor.shared_randomstreams import RandomStreams
```

Comment:
Yes, because the old interface is actually a module and requires your code to be in a module to work.  Since we are deleting modules, we should not keep the old tensor.randomstreams.

I replaced it with shared_randomstreams since that will actually work, we can make an abstract version after.